src/main/scala/com/microsoft/sqlserver/jdbc/spark/connectors/ReliableSingleInstanceStrategy.scala

```diff
@@ -36,7 +36,7 @@ object ReliableSingleInstanceStrategy extends  DataIOStrategy with Logging {
    * Phase 2 Driver combines all staging tables to transactionally write to user specified table.
```

Comment:
Super important to add Tests. Tests need to be added as part of every PR please. 
https://github.com/microsoft/sql-spark-connector/tree/2bf865b906fc036a52332c670fdfc7224576b9ab/test/scala_test

Reply:
I added one test case for both BEST_EFFORT and NO_DUBLICATES. I was trying to change the entire single master instance test frame to 1.run all BEST_EFFOET tests 2. run all NO_DUPLICATES tests 3.run all NO_DUPLICATES tests in parallel. But in parallel has to write to at least sql table, I think it might not necessary to modify every single test case to enable parallel test. Thus I only added one test case for parallel writing.

Reply:
if in parallel, the table names should not be the same. Thus each test case should has table name as input, and then have a corresponding helper function to decide how many thread each function requires.